services/system/HttpServer/src/HttpServer.cpp

```diff
@@ -95,10 +95,26 @@ namespace SystemService
                                                      "Content-Security-Policy",  //
                                                      "ETag"};
 
-      void sendReplyImpl(AccountNumber service, std::int32_t socket, const HttpReply& result)
+      void sendReplyImpl(AccountNumber      service,
+                         std::int32_t       socket,
+                         const HttpReply&   result,
+                         const std::string& requestTarget = "")
       {
          for (const auto& header : result.headers)
          {
+            // Special case: Allow Set-Cookie header only for /common/set-cookie endpoint
+            if (header.name == "Set-Cookie")
+            {
+               if (requestTarget != "/common/set-cookie")
+               {
+                  abortMessage("service " + service.str() +
+                               " attempted to set Set-Cookie header on unauthorized endpoint " +
```

Comment:
Why should only the `/common/set-cookie` endpoint be allowed to set cookies?

Reply:
This was the pre-existing situation; we didn't allow WASMs to set security-related anything, including cookies. I'm just maintaining that, according to discussions we had on this topic. this code opens that window only as far as necessary.

Reply:
I know we discussed that eventually the `/common/set-cookie` endpoint will not have permissive CORS settings (some time after #1274 ), so it can only be called by an app on itself, or by the supervisor.

But I don't think we need to restrict only this endpoint to set cookies.

Reply:
I'd propose until we see that use case or do some research, we just leave this door closed. In fact, how about we just create a separate issue to investigate this?